### PR TITLE
issue210: fix issue with handling of UnknownType in EqualsVisitor

### LINKED ISSUE #210: ClassCastException on equals() of two ClassOrInterfaceDeclaration
```
java.lang.ClassCastException: com.github.javaparser.ast.type.UnknownType cannot be cast to com.github.javaparser.ast.type.WildcardType
    at com.github.javaparser.ast.visitor.EqualsVisitor.visit(EqualsVisitor.java:697)
    at com.github.javaparser.ast.visitor.EqualsVisitor.visit(EqualsVisitor.java:78)
    at com.github.javaparser.ast.type.UnknownType.accept(UnknownType.java:41)
    at com.github.javaparser.ast.visitor.EqualsVisitor.nodeEquals(EqualsVisitor.java:142)
    at com.github.javaparser.ast.visitor.EqualsVisitor.visit(EqualsVisitor.java:538)
    at com.github.javaparser.ast.visitor.EqualsVisitor.visit(EqualsVisitor.java:78)
    at com.github.javaparser.ast.body.Parameter.accept(Parameter.java:60)
    at com.github.javaparser.ast.visitor.EqualsVisitor.nodeEquals(EqualsVisitor.java:142)
    at com.github.javaparser.ast.visitor.EqualsVisitor.nodesEquals(EqualsVisitor.java:117)
    at com.github.javaparser.ast.visitor.EqualsVisitor.visit(EqualsVisitor.java:1433)
    at com.github.javaparser.ast.visitor.EqualsVisitor.visit(EqualsVisitor.java:78)
    at com.github.javaparser.ast.expr.LambdaExpr.accept(LambdaExpr.java:78)
    at com.github.javaparser.ast.visitor.EqualsVisitor.nodeEquals(EqualsVisitor.java:142)
    at com.github.javaparser.ast.visitor.EqualsVisitor.nodesEquals(EqualsVisitor.java:117)
    at com.github.javaparser.ast.visitor.EqualsVisitor.visit(EqualsVisitor.java:985)
    at com.github.javaparser.ast.visitor.EqualsVisitor.visit(EqualsVisitor.java:78)
    at com.github.javaparser.ast.expr.MethodCallExpr.accept(MethodCallExpr.java:67)
    at com.github.javaparser.ast.visitor.EqualsVisitor.nodeEquals(EqualsVisitor.java:142)
    at com.github.javaparser.ast.visitor.EqualsVisitor.visit(EqualsVisitor.java:1223)
    at com.github.javaparser.ast.visitor.EqualsVisitor.visit(EqualsVisitor.java:78)
    at com.github.javaparser.ast.stmt.ExpressionStmt.accept(ExpressionStmt.java:49)
    at com.github.javaparser.ast.visitor.EqualsVisitor.nodeEquals(EqualsVisitor.java:142)
    at com.github.javaparser.ast.visitor.EqualsVisitor.nodesEquals(EqualsVisitor.java:117)
    at com.github.javaparser.ast.visitor.EqualsVisitor.visit(EqualsVisitor.java:1199)
    at com.github.javaparser.ast.visitor.EqualsVisitor.visit(EqualsVisitor.java:78)
    at com.github.javaparser.ast.stmt.BlockStmt.accept(BlockStmt.java:51)
    at com.github.javaparser.ast.visitor.EqualsVisitor.nodeEquals(EqualsVisitor.java:142)
    at com.github.javaparser.ast.visitor.EqualsVisitor.visit(EqualsVisitor.java:515)
    at com.github.javaparser.ast.visitor.EqualsVisitor.visit(EqualsVisitor.java:78)
    at com.github.javaparser.ast.body.MethodDeclaration.accept(MethodDeclaration.java:108)
    at com.github.javaparser.ast.visitor.EqualsVisitor.nodeEquals(EqualsVisitor.java:142)
    at com.github.javaparser.ast.visitor.EqualsVisitor.nodesEquals(EqualsVisitor.java:117)
    at com.github.javaparser.ast.visitor.EqualsVisitor.visit(EqualsVisitor.java:283)
    at com.github.javaparser.ast.visitor.EqualsVisitor.visit(EqualsVisitor.java:78)
    at com.github.javaparser.ast.body.ClassOrInterfaceDeclaration.accept(ClassOrInterfaceDeclaration.java:80)
    at com.github.javaparser.ast.visitor.EqualsVisitor.nodeEquals(EqualsVisitor.java:142)
    at com.github.javaparser.ast.visitor.EqualsVisitor.equals(EqualsVisitor.java:83)
    at com.github.javaparser.ast.Node.equals(Node.java:248)
...
```

when calling `equals()` on two different parses (but both were parsed as ClassOrInterfaceDeclaration) of:

``` java
public class ArrayListGenericDemo {

    public static void main(String[] args) {
        ArrayList<String> data = new ArrayList();
        data.add("hello");
        data.add("goodbye");
        data.forEach( s -> System.out.println(s));
    }
}
```

Parses were obtained directly from `ASTParser.AnnotationBodyDeclaration()` and `ASTParser.ClassOrInterfaceBodyDeclaration(false)` through reflection.

version 2.2.2

### PR DESCRIPTION
**Pull request by ftomassetti**
Fix #210 and add a test to prevent regressions

### CODE REVIEW

File: javaparser-testing/src/test/java/com/github/javaparser/bdd/steps/ComparingSteps.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        this.second = JavaParser.parse(new StringReader(classSrc.trim()), true);
+    }
+
+    /*
+     * Then steps
+     */
+
+    @Then("they are equals")
+    public void thenTheyAreEquals() {
+        assertEquals("they are equals", first, second);
```
Comment (SmiddyPence) [inline javaparser-testing/src/test/java/com/github/javaparser/bdd/steps/ComparingSteps.java]: Would prefer this as Hamcrest assertThat for consistency.

**Comment by SmiddyPence** [comment]
+1 Ship

File: javaparser-testing/src/test/java/com/github/javaparser/bdd/steps/ComparingSteps.java (hunk 1 above)
Comment (ftomassetti) [inline javaparser-testing/src/test/java/com/github/javaparser/bdd/steps/ComparingSteps.java]: Updated

**Comment by ftomassetti** [comment]
Using Hamcrest matchers as suggested. Tests ran again and passing. Merging.

**Pull request merged**